ArcadiaTeamsBot/Dialogs/OpenedRequestsDialog.cs

```diff
@@ -4,36 +4,74 @@
     using System.Threading.Tasks;
 
+    using AdaptiveCards;
+
+    using ArcadiaTeamsBot.CQRS.Abstractions;
+
+    using MediatR;
+
     using Microsoft.Bot.Builder;
     using Microsoft.Bot.Builder.Dialogs;
     using Microsoft.Bot.Schema;
 
     public class OpenedRequestsDialog : ComponentDialog
     {
         private const string Back = "Back";
+        const string Username = "vyacheslav.lasukov@arcadia.spb.ru";
+        private readonly IMediator mediator;
 
-        public OpenedRequestsDialog() : base(nameof(OpenedRequestsDialog))
+        public OpenedRequestsDialog(IMediator mediator) : base(nameof(OpenedRequestsDialog))
         {
+            this.mediator = mediator;
+
             this.AddDialog(new WaterfallDialog(nameof(WaterfallDialog), new WaterfallStep[]
             {
-                TypeStep,
+                this.InfoStep,
                 EndStep,
             }));
             this.AddDialog(new TextPrompt(nameof(TextPrompt)));
             this.InitialDialogId = nameof(WaterfallDialog);
         }
 
-        private static async Task<DialogTurnResult> TypeStep(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        private async Task<DialogTurnResult> InfoStep(WaterfallStepContext stepContext, CancellationToken cancellationToken)
         {
-            var attachments = new[]
+            var getRequestsQuery = new GetCurrentServiceDeskRequestsQuery(Username);
+            var openedRequest = await this.mediator.Send(getRequestsQuery, cancellationToken);
+
+            var Actions = new List<AdaptiveAction>();
+            foreach (var request in openedRequest)
             {
-                GetInfoCard().ToAttachment()
-            };
-            return await stepContext.PromptAsync(nameof(TextPrompt),
-                new PromptOptions
+                if (request.ExecutorFullName == "")
+                {
+                    request.ExecutorFullName = "-";
+                }
+
+                var shownCard = new AdaptiveShowCardAction()
                 {
-                    Prompt = (Activity)MessageFactory.Attachment(attachments),
-                }, cancellationToken);
+                    Title = request.RequestNumber,
+                    Card = new AdaptiveCard()
+                    {
+                        Body = new List<AdaptiveElement>()
+                        {
+                            new AdaptiveTextBlock() { Text = request.Title, Weight = AdaptiveTextWeight.Bolder, Size = AdaptiveTextSize.Large },
+                            new AdaptiveTextBlock() { Text = request.StatusName + ": " + request.Created.ToShortDateString(), Weight = AdaptiveTextWeight.Default, Size = AdaptiveTextSize.Medium },
+                            new AdaptiveTextBlock() { Text = "Executor Name: " + request.ExecutorFullName, Weight = AdaptiveTextWeight.Normal, Size = AdaptiveTextSize.Default },
+                        }
+                    },
+                };
```

Comment:
You can move this all to separate method like `GetRequestAction`, this will make the method clearer.

Reply:
Moved